src/app/components/categories-table/categories-table.component.ts

```diff
@@ -9,69 +9,87 @@ import { Router } from '@angular/router';
 import { RouteValues } from 'src/app/models/constants';
 
 @Component({
-  selector: 'app-categories-table',
-  templateUrl: './categories-table.component.html',
-  styleUrls: ['./categories-table.component.scss']
+	selector: 'app-categories-table',
+	templateUrl: './categories-table.component.html',
+	styleUrls: ['./categories-table.component.scss'],
 })
 export class CategoriesTableComponent implements OnInit {
+	dataSource: Category[] = [];
+	displayedCategories: CategoryToShow[] = [];
+	displayedColumns = ['Icon', 'Title', 'Actions'];
+	valoare: Number = 0;
+	deleteRow: boolean = false;
+	constructor(
+		private endpointsService: EndpointsService,
+		private sanitizer: DomSanitizer,
+		private dialogService: DialogService,
+		private router: Router
+	) {}
 
-  dataSource: Category[] = [];
-  displayedCategories: CategoryToShow[] = [];
-  displayedColumns = ["Icon", "Title", "Actions"];
-  valoare: Number = 0;
-  constructor(
-    private endpointsService: EndpointsService, 
-    private sanitizer: DomSanitizer, 
-    private dialogService: DialogService,
-    private router: Router) {
-  }
-  
-  ngOnInit(): void {
-    this.endpointsService.getCategories().subscribe(categories => {
-      this.dataSource = categories.result;
-      this.createCategoriesToShow();
-    });
-  }
+	ngOnInit(): void {
+		this.endpointsService.getCategories().subscribe((categories) => {
+			this.dataSource = categories.result;
+			this.createCategoriesToShow();
+		});
+	}
 
-  createCategoriesToShow(): void {
-    for (const category of this.dataSource) {
-      let newCategory: CategoryToShow = {
-        id: category.id,
-        title: category.title,
-        icon: this.sanitizer.bypassSecurityTrustResourceUrl(category.icon),
-        defaultBackground: this.sanitizer.bypassSecurityTrustResourceUrl(category.defaultBackground),
-      }
-      this.displayedCategories.push(newCategory);
-    }
-  }
+	createCategoriesToShow(): void {
+		for (const category of this.dataSource) {
+			let newCategory: CategoryToShow = {
+				id: category.id,
+				title: category.title,
+				icon: this.sanitizer.bypassSecurityTrustResourceUrl(
+					category.icon
+				),
+				defaultBackground:
+					this.sanitizer.bypassSecurityTrustResourceUrl(
+						category.defaultBackground
+					),
+			};
+			this.displayedCategories.push(newCategory);
+		}
+	}
 
-  deleteCategory(categoryId: number): void {
-    this.openChangeRoleDialog().subscribe(result => {
-      if (result)
-        this.endpointsService.deleteCategory(categoryId)
-          .subscribe(() => {
-            this.displayedCategories = this.displayedCategories
-              .filter((category: CategoryToShow) => category.id != categoryId)
-          });
-    })
-  }
+	deleteCategory(categoryId: number): void {
+		this.deleteRow = true;
```

Comment:
The delete category button belongs to the row of the category, so clicking it also redirects you to that specific category's page. So I have made it so that when clicking the delete button a switch is triggered, and if said switch is true, navigation to the category page does not occur.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault
prevent the default browser behavior on the click event for delete icon :)